python/graphstorm/gconstruct/transform.py

```diff
@@ -52,6 +53,148 @@ def feat_name(self):
         """
         return self._feat_name
 
+class TwoPhaseFeatTransform(FeatTransform):
+    """ The base class for two phasefeature transformation.
+
+        The first phase is going to collect global information
+        for data processing, e.g., collecting max and min value of floating
+        point data or collecting the cardinality of categorical data.
+        The second phase is to transform data using
+        the global
+        information collected in the first phase
+
+    Parameters
+    ----------
+    col_name : str
+        The name of the column that contains the feature.
+    feat_name : str
+        The feature name used in the constructed graph.
+    """
+
+    def pre_process(self, feats):
+        """ Pre-process data
+
+        Parameters
+        ----------
+        feats:
+            feats to be processed
+        """
+
+    def collect_info(self, info):
+        """ Store global information for the second phase data processing
+
+        Parameters
+        ----------
+        info:
+            Information to be collected
+        """
+
+class NumericalTransform(TwoPhaseFeatTransform):
```

Comment:
this name sounds too general. if this is doing min max, it should be named with MinMaxTransform

Reply:
It is a template class for numerical transform ops. It only collects max/min info. Any numerical transform ops which needs the max/min info can inherit from NumericalTransform.
NumericalMinMaxTransform is for max/min normalization.

Reply:
i'm not sure if this kind of abstraction is needed.

Reply:
what it does is to collect min and max. it makes more sense to just have `NumericalMinMaxTransform`

Reply:
do you have any other examples that this base class can be used for other operations?